internal/cloudapi/v2/handler.go

```diff
@@ -212,6 +212,15 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
 			return err
 		}
 
+		// If no repositories are included with the imageRequest use the defaults for the distro
+		if len(ir.Repositories) == 0 {
+			dr, err := h.server.repos.ReposByImageTypeName(distribution.Name(), arch.Name(), imageType.Name())
+			if err != nil {
+				return err
+			}
+			repos = append(repos, dr...)
+		}
```

Comment:
I think this is ok as is, an empty repositories list is still required (without it openapi returns an error because of the missing keyword) so it's pretty clear what the intention is. And I hate adding more switches :)